### PR TITLE
add(): The ability to load item states for bedrock edition

### PR DESCRIPTION
**Pull request by ATXLtheAxolotl**
You will probably want to  squash this PR.

### CODE REVIEW

File: lib/bedrock/index.js [hunk 1]
```diff
@@ -1 +1,27 @@
...
+              ..._item,
+              id: name === _item.name ? item.runtime_id : _item.id
+            }
+          })
+        }
+      }
+      data.items = this.buildIndexFromArray(data.itemsArray, 'id')
+      data.itemsByName = this.buildIndexFromArray(data.itemsArray, 'name')
+    },
+    buildIndexFromArray (array, fieldToIndex) {
```
Comment (extremeheat) [inline lib/bedrock/index.js]: Should not be exposed

**Review by extremeheat — CHANGES_REQUESTED** [review]
Missing docs, tests and writing

File: lib/bedrock/index.js (hunk 1 above)
Comment (ATXLtheAxolotl) [reply lib/bedrock/index.js]: How would I make `buildIndexFromArray` private in JavaScript?

Comment (extremeheat) [reply lib/bedrock/index.js]: Put it somewhere else

File: lib/bedrock/index.js [hunk 2]
```diff
@@ -1 +1,22 @@
-module.exports = (data) => ({})
+const { buildIndexFromArray } = require('./indexer')
+
+module.exports = (data) => {
+  return {
+    loadItemStates (itemStates) {
+      for (const item of itemStates) {
+        const name = item.name.replace('minecraft:', '')
+
+        if (data.itemsArray) {
```
Comment (extremeheat) [inline lib/bedrock/index.js]: This seems pretty inefficient (double for loop), can you explain what you're doing here?

**Review by extremeheat — CHANGES_REQUESTED** [review]

File: lib/bedrock/index.js (hunk 2 above)
Comment (ATXLtheAxolotl) [reply lib/bedrock/index.js]: So basically it loops over the item states, and for each item in there it gets the name, then change the id of the existing item with the same name stored in `itemsArray`.

Comment (extremeheat) [reply lib/bedrock/index.js]: why not just something like this:
```js
const items = []
for (const item of itemStates) {
  const name = item.name.replace('minecraft:', '')
  items.push({ name, ...staticData.itemsByName[name], id: item.runtime_id })
}
data.itemsArray = items
...
```
?

Comment (ATXLtheAxolotl) [reply lib/bedrock/index.js]: I didn't think of that

**Comment by extremeheat** [comment]
Missing:
* tests
* analog function to write the palette

**Comment by rom1504** [comment]
what is the status here

**Comment by ATXLtheAxolotl** [comment]
I’m not sure; PR #26 expanded on this one’s work, they were talking about making a write function

**Comment by rom1504** [comment]
closing in favor of #26

**Pull request closed without merging**